src/wallet/walletdb.cpp

```diff
@@ -345,7 +346,13 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             // LoadToWallet call below creates a new CWalletTx that fill_wtx
             // callback fills with transaction metadata.
             auto fill_wtx = [&](CWalletTx& wtx, bool new_tx) {
-                assert(new_tx);
+                if(!new_tx) {
+                    // There's some corruption here since the tx we just tried to load was already in the wallet.
+                    // We don't consider this type of corruption critical, and can fix it by removing tx data and
+                    // rescanning.
```

Comment:
Probably I am missing context here, so feel free to ignore my question.

Why would we consider "the tx we just tried to load was already in the wallet." as a corrupted transaction? If I am missing context here, would it be helpful to clarify this in the above comment?

Reply:
> Why would we consider "the tx we just tried to load was already in the wallet." as a corrupted transaction? If I am missing context here, would it be helpful to clarify this in the above comment?

It might not be bad to update the comment above or the error message below or the `tx_corrupt` variable name if you have suggestions, but this all seems ok to me. If there are multiple transactions with the same hash, the wallet is corrupt, and the transactions are maybe corrupt. In this case it is probably better to not load the wallet than to load it and make new writes, and to return a loading error instead of asserting false.

Reply:
Hmm. The context I was missing then would be: `!new` suggests that the wallet already loaded the transaction with the given hash.

So maybe the comment above could read something like "There's some corruption here since the wallet already knows about a transaction with the same Txid". That would clarify why this case is an error a little clearly.